transactions/models.py

```diff
@@ -0,0 +1,44 @@
+from django.db import models
+
+# Create your models here.
+from blocks.models import Block
+
+
+class Transaction(models.Model):
+    hash = models.CharField(max_length=100)
+    block = models.ForeignKey(Block, blank=True, null=True, on_delete=models.RESTRICT, related_name='transactions')
+    tx_ins_ct = models.IntegerField(blank=True, default=0)
+    tx_outs_ct = models.IntegerField(blank=True, default=0)
+    mined = models.BooleanField(default=False)
+
+    def __str__(self):
+        return f"tx hash {self.hash} in block {self.block}"
+
+
+class TransactionComponent(models.Model):
+    hash = models.CharField(max_length=100, blank=True, null=True)
+    value = models.IntegerField(default=0)
+    own_addr = models.CharField(max_length=100)
+    gen_tx_idx = models.IntegerField(blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class TxIn(TransactionComponent):
+    gen_tx = models.ForeignKey(Transaction, blank=True, null=True, on_delete=models.RESTRICT)
+    spend_tx = models.ForeignKey(Transaction, blank=True, null=True, on_delete=models.RESTRICT, related_name='inputs')
```

Comment:
Well first we chose Restrict since we wanted to not delete anything at all in our db. Now that we think of it we may need to make changes in the blockchain depending on the information we receive from nodes. In such a case we think that e.g if we delete a block, its transactions should also go away... if we delete a transactions its TxIns and TxOuts should go away.... hence we have changes models.RESTRICT to models.CASCADE